src/vm/runners/builtin_runner/mod.rs

```diff
@@ -66,6 +66,22 @@ impl BuiltinRunner {
         }
     }
 
+    pub fn final_stack(
```

Comment:
Also, It would be useful to have this decision explained in the PR description

Reply:
I don't think the explanation is really needed. AFAICT the project is not about being a verbatim translation across languages but about being a drop-in replacement. If a solution is cleaner for Rust and obvious enough (and a pure function here is) I don't think it merits extra explanation.

Reply:
Even if its not meant to be a drop-in replacement, our current approach to the Starknet integration relies on exposing the methods expected from the original implementation, so even if we think updating the stop_ptr is not the responsibility of this method we should at least leave a comment so that we are aware of this when calling the method and don't run into unexpected bugs when we run code that expects the stop_ptr to have a certain value after calling final_stack.

Reply:
Explanation is not necessary, just nice to have for reference, but I still think comments are needed for this (mostly to prevent unexpected bugs later)

Reply:
AFAIU this method is not used directly at a higher level than its caller in cairo-rs-py, but fair enough.